library/include/updater.h

```diff
@@ -14,6 +14,12 @@
 #endif
 
 
+typedef enum UpdateBehavior {
+  BackgroundOnLaunch,
+  Manual,
+  WaitOnLaunch,
```

Comment:
```suggestion
  AsyncOnLaunch,
  SyncOnLaunch,
  Manual,
```

Reply:
feel free to ignore but tried to come up with more consistent names for the modes 🤷

Reply:
Sure. 🤷   Naming is the hardest part here.  Those are the internal enum names, but we also have to figure out the public yaml names (and the yaml property name).

Reply:
Yeah the public facing names are more important especially because it'll be tough to change after 🤷

Reply:
So just to make sure I follow the modes we want to support are:

1. Check for updates on launch and install in background (`background`, what we currently do)
2. Check for updates on launch and install blocking (`wait`)
3. Do not check for updates, leave it up to the developer (`manual`)

Reply:
What if we split this into `update_check_behavior` and `update_installation_behavior`?

```yaml
update_check_behavior: launch
update_install_behavior: blocking | non-blocking
```

Reply:
Also the more I think about it the more I wonder whether or not we should just have a flag:

```yaml
auto_update: true | false
```

For all custom scenarios we could just leave it up to `package:shorebird_code_push`